src/utils/api/account/lsk.js

```diff
@@ -79,26 +80,37 @@ export const getAccount = async ({
       params: normParams,
     });
 
-    return response.data[0];
-  } catch (e) {
-    // eslint-disable-next-line no-console
-    console.log('Lisk account not found.');
-    let publicKey = params.publicKey;
-    if (!publicKey && params.passphrase) {
-      publicKey = extractPublicKey(params.passphrase);
+    if (response.data[0]) {
+      const account = { ...response.data[0] };
+      const isAccountUninitialized = !account.summary.publicKey;
+      if (isAccountUninitialized && (params.publicKey || params.passphrase)) {
+        const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
+        account.summary.publicKey = publicKey;
+      }
+      return account;
     }
+  } catch (e) {
+    if (e.code === HTTP_CODES.NOT_FOUND) {
+      // eslint-disable-next-line no-console
+      console.log('Lisk account not found.');
 
-    const account = {
-      summary: {
-        publicKey,
-        balance: 0,
-        address: normParams.address,
-        token: tokenMap.LSK.key,
-      },
-    };
-
-    return account;
+      if (params.publicKey || params.passphrase) {
+        const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
+        const account = {
+          summary: {
+            publicKey,
+            balance: 0,
+            address: normParams.address,
+            token: tokenMap.LSK.key,
+          },
+        };
+        return account;
+      }
+    } else {
+      throw Error();
```

Comment:
This error statement needs a message

Reply:
The text would be the same as the outer error which actually catches this error (and has the required text) so I did not add any text.

Reply:
Yes. I meant it seems like you've done a good job covering all the scenarios. Even at the event of network failure as you said in the next comment, it would end up here and this error will suffice. but not a big deal, I'll approve it.